cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+    case 'PPL':
+      return 'PPL';
+    default:
+      return 'unknown language';
+  }
+};
+
+// Escapes special characters for the editor
+const prepareQueryStringForEditor = (queryString) => {
+  return queryString.replaceAll(/([{}])/g, (char) => `{${char}}`);
+};
+
+// In theory this function is not needed as we should be able to use
+// the custom command navigateToWorkSpaceSpecificPage, but when using that
+// to go to the discover page, it sometimes leads to a blank page
+const navigateToDiscoverPage = () => {
+  cy.navigateToWorkSpaceHomePage(workspaceName);
+  cy.getElementByTestId('headerAppActionMenu').should('be.visible');
+};
+
+// Sets the dataset for the search. Since INDEXES are not saved to the dataset,
+// we need to click through various buttons to manually add them
+const setDataset = (dataset) => {
+  const datasetType = mapDatasetToType(dataset);
+  cy.getElementByTestId('datasetSelectorButton').should('be.visible').click();
+
+  if (datasetType === 'INDEX_PATTERN') {
+    cy.get(`[title="${dataset}"]`).click();
+  } else if (datasetType === 'INDEXES') {
+    cy.getElementByTestId('datasetSelectorAdvancedButton').click();
+    cy.contains('span', 'Indexes').click();
+    cy.contains('span', datasourceName).click();
+    // this element is sometimes being masked by another element
+    cy.contains('span', dataset).should('be.visible').click({ force: true });
+    cy.getElementByTestId('datasetSelectorNext').click();
+
+    cy.getElementByTestId('advancedSelectorTimeFieldSelect').select('timestamp');
+    cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+    cy.getElementByTestId('datasetSelectorButton').should(
+      'contain.text',
+      `${datasourceName}::${dataset}`
+    );
+  }
+};
+
+const setDatePickerDatesAndSearchIfRelevant = (language) => {
+  if (language === 'OpenSearch SQL') {
+    return;
+  }
+
+  cy.setTopNavDate(START_TIME, END_TIME);
+};
+
+const setSearchConfigurations = ({
+  addFilter,
+  queryString,
+  setHistogramInterval,
+  selectFields,
+  applySort,
+}) => {
+  if (addFilter) {
+    cy.getElementByTestId('showFilterActions').click();
+    cy.getElementByTestId('addFilters').click();
+    cy.getElementByTestId('filterFieldSuggestionList').find('input').type('category');
+    cy.getElementByTestId('comboBoxOptionsList filterFieldSuggestionList-optionsList')
+      .find('button[title="category"]')
+      .click();
+    cy.getElementByTestId('filterOperatorList').find('input').type('is one of');
+    cy.getElementByTestId('comboBoxOptionsList filterOperatorList-optionsList')
+      .find('button[title="is one of"]')
+      .click();
+    cy.getElementByTestId('filterParams').find('input').type('Application');
+    cy.getElementByTestId(
+      'comboBoxOptionsList filterParamsComboBox phrasesParamsComboxBox-optionsList'
+    )
+      .find('button[title="Application"]')
+      .click();
+    // Need to wait here a bit to avoid cypress flakiness
+    cy.wait(750);
```

Comment:
what is the flakiness and error here? is it due to load the discover page after add filter? if so, we can use cy.waitForLoader(true)

Reply:
There is no loader here, but sometimes when i click on the "Application" dropdown, it doesn't register right away, causing some error when attempting to set the filter. There's no loader here so i'm not sure if `waitForLoader()` is appropriate

Reply:
This is no longer relevant as I am using a utility